api/v1/jats/PKPJatsController.php

```diff
@@ -0,0 +1,214 @@
+ *
+ * @class PKPJatsController
+ *
+ * @ingroup api_v1_jats
+ *
+ * @brief Handle API requests for JATS File operations.
+ *
+ */
+
+namespace PKP\API\v1\jats;
+
+use APP\core\Application;
+use APP\facades\Repo;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Http\Request;
+use Illuminate\Http\Response;
+use Illuminate\Support\Facades\Route;
+use PKP\core\PKPBaseController;
+use PKP\core\PKPRequest;
+use PKP\db\DAORegistry;
+use PKP\security\authorization\ContextAccessPolicy;
+use PKP\security\authorization\internal\SubmissionFileStageAccessPolicy;
+use PKP\security\authorization\PublicationAccessPolicy;
+use PKP\security\authorization\PublicationWritePolicy;
+use PKP\security\authorization\SubmissionFileAccessPolicy;
+use PKP\security\authorization\UserRolesRequiredPolicy;
+use PKP\security\Role;
+use PKP\services\PKPSchemaService;
+use PKP\submissionFile\SubmissionFile;
+
+class PKPJatsController extends PKPBaseController
+{
+    /**
+     * @copydoc \PKP\core\PKPBaseController::getHandlerPath()
+     */
+    public function getHandlerPath(): string
+    {
+        return 'submissions/{submissionId}/publications/{publicationId}/jats';
+    }
+
+    /**
+     * @copydoc \PKP\core\PKPBaseController::getRouteGroupMiddleware()
+     */
+    public function getRouteGroupMiddleware(): array
+    {
+        return [
+            'has.user',
+            'has.context',
+        ];
+    }
+
+    public function getGroupRoutes(): void
+    {
+        Route::middleware([
+            self::roleAuthorizer([
+                Role::ROLE_ID_MANAGER,
+                Role::ROLE_ID_SITE_ADMIN,
+                Role::ROLE_ID_SUB_EDITOR,
+                Role::ROLE_ID_ASSISTANT,
+                Role::ROLE_ID_AUTHOR,
+            ]),
+        ])->group(function () {
+
+            Route::get('', $this->get(...))
+                ->name('publication.jats.get');
+
+            Route::post('', $this->add(...))
+                ->name('publication.jats.add');
+
+            Route::delete('', $this->delete(...))
+                ->name('publication.jats.delete');
+
+        })->whereNumber(['submissionId', 'publicationId']);
+    }
+
+    /**
+     * @copydoc \PKP\core\PKPBaseController::authorize()
+     */
+    public function authorize(PKPRequest $request, array &$args, array $roleAssignments): bool
+    {
+        $illuminateRequest = $args[0]; /** @var \Illuminate\Http\Request $illuminateRequest */
+        $actionName = static::getRouteActionName($illuminateRequest);
+
+        $this->addPolicy(new UserRolesRequiredPolicy($request), true);
+
+        $this->addPolicy(new ContextAccessPolicy($request, $roleAssignments));
+
+        $this->addPolicy(new PublicationWritePolicy($request, $args, $roleAssignments));
+
+        if ($actionName === 'add') {
+            $params = $illuminateRequest->input();
+            $fileStage = isset($params['fileStage']) ? (int) $params['fileStage'] : 21;
```

Comment:
File stage shouldn't be hard-coded (`21`).

Reply:
Changed to `SubmissionFile::SUBMISSION_FILE_JATS`